### PR TITLE
closes #152

### LINKED ISSUE #152: `0` does not go to beginning of line in normal mode
`^` can be used to go to the first non-whitespace character in the line. `0` and `^` both work correctly in visual mode.

### PR DESCRIPTION
**Pull request by ninrod**
This closes bug #152. I've been using this fork and it's all good. What you think?

### CODE REVIEW

**Comment by ninrod** [comment]
I really don't know why that test is failling. I type the same command sequence and the plugin operates as expected. I cannot reproduce that failed result.

**Comment by eapache** [comment]
Thanks for this, but I think this fix is incomplete; we do want to immediately consume `0` if it is the first digit, but not if there are preceding digits - otherwise you can't enter commands like `10j` to move ten lines down.

**Comment by ninrod** [comment]
@eapache Thanks for the insight. I think I've managed to work my way around this scenario now. 
I `10j`and alikes are working normally now in normal, visual and visual line modes.

File: autoload/multiple_cursors.vim [hunk 1]
```diff
@@ -1207,8 +1207,16 @@ function! s:wait_for_user_input(mode)
     call feedkeys(s:char)
     call s:cm.reset(1, 1)
     return
-  elseif s:from_mode ==# 'n'
+  elseif s:from_mode ==# 'n' || s:from_mode =~# 'v\|V'
```
Comment (eapache) [inline autoload/multiple_cursors.vim]: this now covers all modes but insert mode right? can we just make it `!=# 'i'`?

**Comment by eapache** [comment]
Looks good to me, but I can't test it for the next few days so I'll let somebody else hit the merge button.

File: autoload/multiple_cursors.vim (hunk 1 above)
Comment (ninrod) [inline autoload/multiple_cursors.vim]: I don't think it covers all modes except `i`.  It just covers `i, v and V`. I think we should not add this behaviour to command mode and replace mode, besides insert mode. (from the top of my head)

**Comment by ninrod** [comment]
@eapache @faceleg I'd like to add some tests for this pull but I have no idea how to run them locally. I understand they use ruby, but that's about it.

**Comment by balta2ar** [comment]
@ninrod It's something along these lines:
``` bash
# install ruby
# install rake
gem install rake
# install vimrunner - ruby "wrapper" for vim
gem install vimrunner
# run rake in the root directory
cd vim-multiple-cursors
rake
```

File: autoload/multiple_cursors.vim (hunk 1 above)
Comment (eapache) [inline autoload/multiple_cursors.vim]: Oh yes you're right I'm not sure what I was thinking forgetting command mode :)

**Comment by eapache** [comment]
@ninrod @balta2ar it's actually simpler than that if you install ruby and bundler first. In the root of the repository:
```
bundle install
bundle exec rake
```

**Comment by ninrod** [comment]
Thanks gentleman. I'll try to do that when I get spare time. We should add a quick-start guide for setting up the tests.

**Comment by balta2ar** [comment]
Thanks! Maybe it's worth adding to README.md for future contributors.

**Pull request closed without merging**